modules/programs/firefox.nix

```diff
@@ -8,6 +8,26 @@ let
 
   extensionPath = "extensions/{ec8030f7-c20a-464f-9b0e-13a3a9e97384}";
 
+  profiles = (mapAttrs' (_: profile: nameValuePair "Profile${toString profile.id}" {
+    Name = profile.name;
+    Path = profile.path;
+    IsRelative = 1;
+    Default = if profile.isDefault then 1 else 0;
+  }) cfg.profiles) // {
+    General = {
+      StartWithLastProfile = 1;
+    };
+  };
+
+  profilesINI = generators.toINI {} profiles;
+
+  mkUserJS = prefs: extraPrefs: ''
+    ${concatStrings (mapAttrsToList (name: value: ''
+    user_pref("${name}", ${builtins.toJSON value});
```

Comment:
I assume the configuration name is unlikely to need escaping, otherwise could have
```
user_pref("${builtins.toJSON name}", ${builtins.toJSON value});
```

Reply:
It's already quoted, so it should be fine, i do not think `builtins.toJSON` is the right way for name.